src/contexts/addresses.tsx

```diff
@@ -353,7 +353,7 @@ export const AddressesContextProvider: FC<{ overrideContextValue?: PartialDeep<A
       if (addressesMetadata.length === 0) {
         saveNewAddress(
           new Address(wallet.address, wallet.publicKey, wallet.privateKey, 0, {
-            isMain: true,
+            isMain: isPassphraseUsed ? false : true,
```

Comment:
I am not so sure about this change. Even though visually we don't have the concept of "main" address for passphrase-enabled wallets, I am afraid that our business logic might break when there is no single main address (all references to `mainAddress` will be broken). I would not make this change.

Reply:
When I was testing it everything seemed ok? I can revert it for now but it might be a good idea to investigate removing  the dependency of requiring a main address...